modules/consentManagement.js

```diff
@@ -47,11 +48,70 @@ function lookupStaticConsentData(cmpSuccess, cmpError, hookConfig) {
  * @param {object} hookConfig contains module related variables (see comment in requestBidsHook function)
  */
 function lookupIabConsent(cmpSuccess, cmpError, hookConfig) {
-  function handleCmpResponseCallbacks() {
+  function findCMP() {
+    let f = window;
+    let cmpFrame;
+    let cmpFunction;
+    while (!cmpFrame) {
+      try {
+        if (typeof f.__tcfapi === 'function' || typeof f.__cmp === 'function') {
+          if (typeof f.__tcfapi === 'function') {
+            cmpVersion = 2;
+            cmpFunction = f.__tcfapi;
+          } else {
+            cmpVersion = 1;
+            cmpFunction = f.__cmp;
+          }
+          cmpFrame = f;
+          break;
+        }
+      } catch (e) { }
+
+      // need separate try/catch blocks due to the exception errors thrown when trying to check for a frame that doesn't exist in 3rd party env
+      try {
+        if (f.frames['__tcfapiLocator']) {
+          cmpVersion = 2;
+          cmpFrame = f;
+          break;
+        }
+      } catch (e) { }
+
+      try {
+        if (f.frames['__cmpLocator']) {
+          cmpVersion = 1;
+          cmpFrame = f;
+          break;
+        }
+      } catch (e) { }
+
+      if (f === window.top) break;
+      f = f.parent;
+    }
+    return {
+      cmpFrame,
+      cmpFunction
+    };
+  }
+
+  function v2CmpResponseCallback(tcfData, success) {
+    utils.logInfo('Received a response from CMP', tcfData);
+    if (success) {
+      if (tcfData.eventStatus === 'tcloaded' || tcfData.eventStatus === 'useractioncomplete') {
+        cmpSuccess(tcfData, hookConfig);
+      } else if (tcfData.eventStatus === 'cmpuishown' && tcfData.tcString.length > 0 && tcfData.purposeOneTreatment === true) {
```

Comment:
This was the advice from Xandr legal counsel. It would be good if there were a way to allow publishers to bypass this check. Flexibility being our philosophy here. @jsnellbaker and @jaiminpanchal27 - would how about we add something like this:

```
pbjs.setConfig({
   consentManagement: {
      gdpr: {
        checkPurposeOneTreatmentOnRead: false    // default to true
      }
   }
});
```

I'll bounce this off Grant.

Reply:
Trying to understand the lawyer's rational - Purpose 1 has to do with device access, right?  And this conditional says that we're passing the consent string (if it exists) through to the bidders while the cmp ui is still being shown to the user, as long as they've given device access consent.  What does passing consent on to the bidders have to do with device access?
Taking a shot at an explanation -  If we know that the user has already given some positive consent (good) pass the string along.  If not (not as good), give them the opportunity to change it.

Reply:
It's saying that we can move on with an older or incomplete version of consent without waiting for the useractioncomplete event as long as the string contains indication that publisher declared the ability to "read the device". It's unclear to me that this is an obvious fit for purposeOneTreatment, but P1T's actual meaning is murky and this doesn't seem like it should hurt anything.

If P1T isn't utilized, the PBJS will just wait for the user action to complete.

Reply:
ok